gobblin-data-management/src/main/java/gobblin/data/management/dataset/DatasetUtils.java

```diff
@@ -63,7 +63,7 @@ public boolean accept(Path path) {
    * @throws IOException
    */
   @SuppressWarnings("unchecked")
-  public static <T extends Dataset> DatasetFinder<T> instantiateDatasetFinder(Properties props, FileSystem fs,
+  public static <T extends gobblin.dataset.Dataset> DatasetFinder<T> instantiateDatasetFinder(Properties props, FileSystem fs,
```

Comment:
Since it has been changed to generic `dataset`, the constructor still expects for `FileSystem`, which implies it to be a `FileBasedDataset`. Are you planning to change `DatasetFinder` to be more generic as well?